tests/integration/cocoon/test_cocoon_transactions.py

```diff
@@ -898,19 +905,30 @@ def test_load_from_dataframe_non_existent_subholding_keys(
             sub_holding_keys=sub_holding_keys
         )
 
+        # we get the details from the portfolio
         portfolio_details = transaction_portfolio_api.get_details(scope=scope, code='no-SHK')
 
+        # get the transactions that we just loaded in
         transactions = transaction_portfolio_api.get_transactions(scope=scope, code='no-SHK')
 
+        # delete the portfolio as it is no longer required
         try:
             portfolios_api.delete_portfolio(scope, 'no-SHK')
         except lusid.ApiException as e:
             if "PortfolioNotFound" not in str(e.body):
                 raise e
 
+        # check that the sub holding key is a property of the two transactions
         self.assertTrue('Transaction/load_dataframe_test/SHK_data' in transactions.values[0].properties.keys())
         self.assertTrue('Transaction/load_dataframe_test/SHK_data' in transactions.values[1].properties.keys())
 
+        # delete the property now that we know that it exists.
+        self.api_factory.build(lusid.PropertyDefinitionsApi).delete_property_definition('Transaction',
```

Comment:
put it in a try except clause with a similar, not found thing as you have done above with the portfolios